extras/splat/splat.js

```diff
@@ -198,8 +198,9 @@ class Splat {
                 data[i * 4 + 2] = math.clamp((0.5 + SH_C0 * c2[i]) * 255, 0, 255);
             }
 
-            // opacity
-            data[i * 4 + 3] = opacity ? math.clamp(sigmoid(opacity[i]) * 255, 0, 255) : 255;
+            
+            // opacity (-inf..+inf, at 0.0 it returns 0.5)
+            data[i * 4 + 3] = (opacity !== undefined) ? sigmoid(opacity[i]) * 255 : 255;
```

Comment:
But this previous code was just checking that the `Float32Array` `opacity` array was defined. Not a _value_ from that array. If anything is wrong in this function, it's the JSDocs. The `opacity` param should be marked as optional presumably?

Reply:
I agree, based on `if (c0 && c1 && c2) {` they should all be optional.

Code is called from here:

https://github.com/playcanvas/engine/blob/f9bac39b27f9df00d1ca6285a3c1c36b5af77fd4/extras/splat/splat-container-resource.js#L50

And I found a PLY that has other prop names aswell (using the prop names 'red', 'green' and 'blue')

Reply:
@willeastcott oh damn, my apologies! The hazards of making a public pull request after a long day... Yes you're correct of course, the opacity condition is an array existence check.
The clamp is definitely not needed though.

Reply:
@pjbaron No worries! 😄 Do you wanna update the code to reflect this?